src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -6638,6 +6672,12 @@ public static String replaceEach(final String text, final String[] searchList, f
      *             and/or size 0)
      * @since 2.4
      */
+    @SuppressWarnings({"index:array.access.unsafe.high","index:argument.type.incompatible","index:array.access.unsafe.low"}) /*
+    #1: searchLength = replacementLength is checked in #0.1
+    #2: textIndex != -1 and is an index searched for in the string, hence a valid index
+    #3: textIndex != -1 => replaceIndex != -1
+    #4: searchLength = replacementLeng as checked in #0.1
+    */
     private static String replaceEach(
             final String text, final String[] searchList, final String[] replacementList, final boolean repeat, final int timeToLive) {
```

Comment:
I would annotate `searchList` and `replacementList` as `@SameLen` of each other following https://checkerframework.org/manual/#annotate-normal-behavior